test/testConfigure.js

```diff
@@ -635,6 +638,128 @@ describe('Configure', function() {
+    configure();
+
+    writeYaml.sync('.travis.yml', {
+      env: {
+        global: [ 'ENV_GLOBAL' ],
+        matrix: [ 'ENV_MATRIX' ],
+      },
+    });
+
+    return enterUsernamePassword()
+    .then(function() {
+      return await('You\'re ready to auto-deploy using Travis!');
+    })
+    .then(function() {
+      var travisYml = readYaml.sync('.travis.yml');
+      expect(travisYml).to.include.keys('env');
+      expect(travisYml.env).to.include.keys('global');
+      expect(travisYml.env).to.include.keys('matrix');
+      expect(travisYml.env.global).to.have.length(2);
+      expect(travisYml.env.global[0]).to.equal('ENV_GLOBAL');
+      expect(travisYml.env.global[1]).to.have.keys('secure');
+      expect(travisYml.env.matrix).to.have.length(1);
+      expect(travisYml.env.matrix[0]).to.equal('ENV_MATRIX');
+    });
+  });
+
+  it('does not remove "before_script" in an already existing .travis.yml file', function() {
+    nockBasicPostAuthFlow();
+    configure();
+
+    writeYaml.sync('.travis.yml', {
+      before_script: [
+        'a_command',
+      ],
+    });
+
+    return enterUsernamePassword()
+    .then(function() {
+      return await('You\'re ready to auto-deploy using Travis!');
+    })
+    .then(function() {
+      var travisYml = readYaml.sync('.travis.yml');
+      expect(travisYml.before_script).to.have.length(3);
+      expect(travisYml.before_script[0]).to.equal('a_command');
+    });
+  });
+
+  it('does not overwrite "before_script" in an already existing .travis.yml file', function() {
+    nockBasicPostAuthFlow();
+    configure();
+
+    writeYaml.sync('.travis.yml', {
+      before_script: [
+        'git config --global user.name "A User"',
+        'git config --global user.email "a_user@mozilla.org"',
+      ],
+    });
+
+    return enterUsernamePassword()
+    .then(function() {
+      return await('You\'re ready to auto-deploy using Travis!');
+    })
+    .then(function() {
+      var travisYml = readYaml.sync('.travis.yml');
+      expect(travisYml.before_script).to.have.length(2);
+      expect(travisYml.before_script[0]).to.equal('git config --global user.name "A User"');
+      expect(travisYml.before_script[1]).to.equal('git config --global user.email "a_user@mozilla.org"');
+    });
+  });
+
+  it('does not remove "after_success" in an already existing .travis.yml file', function() {
+    nockBasicPostAuthFlow();
+    configure();
+
+    writeYaml.sync('.travis.yml', {
+      after_success: [
+        'a_command'
+      ]
+    });
+
+    return enterUsernamePassword()
+    .then(function() {
+      return await('You\'re ready to auto-deploy using Travis!');
+    })
+    .then(function() {
+      var travisYml = readYaml.sync('.travis.yml');
+      expect(travisYml.after_success).to.have.length(2);
+      expect(travisYml.after_success[0]).to.equal('a_command');
+      expect(travisYml.after_success[1]).to.equal(
+        'echo "travis_fold:end:after_success" && ' +
```

Comment:
This is testing what a configuration setting is, which is not the scope of this test. Can we make it a more general assertion?